### DISCUSSION TITLE
What are other ways to "Reboot" a Discord bot using Discord.js?

### DISCUSSION THREAD

**Discussion by goustaff** [discordjs/discord.js#10320] [Q&A]
Hello!
The documentation says:
https://discordjs.guide/popular-topics/faq.html#how-do-i-restart-my-bot-with-a-command

```ts
process.exit();
```
Somehow, that's cute.
But, which alternatives do we have to do it programmatically instead of using sys workarounds?

**Comment by Qjuh** [comment]
What exactly do you expect a „reboot“ of your bot do? If not restarting the node process…

**Reply by goustaff** [reply]
> What exactly do you expect a „reboot“ of your bot do? If not restarting the node process…

I'd expect a program to be resumable without killing a process and restarting it. :/

**Reply by Qjuh** [reply]
Resumable from what? In case of disconnects/reconnects discord.js does that automatically for you, no action required. So what exactly is it you want your „command to reboot the bot“ do? What do you expect to gain from a restart?
